controllers/volumereplicationgroup_controller.go

```diff
@@ -538,65 +538,23 @@ const (
 	MModesLabel = "ramendr.openshift.io/maintenancemodes"
 )
 
-func (v *VRGInstance) processVRG() (ctrl.Result, error) {
+func (v *VRGInstance) processVRG() ctrl.Result {
 	if err := v.validateVRGState(); err != nil {
-		// record the event
-		v.log.Error(err, "Failed to validate the spec state")
-		rmnutil.ReportIfNotPresent(v.reconciler.eventRecorder, v.instance, corev1.EventTypeWarning,
-			rmnutil.EventReasonValidationFailed, err.Error())
-
-		msg := "VolumeReplicationGroup state is invalid"
-		setVRGDataErrorCondition(&v.instance.Status.Conditions, v.instance.Generation, msg)
-
-		if err = v.updateVRGStatus(false); err != nil {
-			v.log.Error(err, "Status update failed")
-			// Since updating status failed, reconcile
-			return ctrl.Result{Requeue: true}, nil
-		}
 		// No requeue, as there is no reconcile till user changes desired spec to a valid value
-		return ctrl.Result{}, nil
+		return v.invalid(err, "VolumeReplicationGroup state is invalid", false)
 	}
 
 	// If neither of Async or Sync mode is provided, then
 	// dont requeue. Just return error.
 	if err := v.validateVRGMode(); err != nil {
-		// record the event
-		v.log.Error(err, "Failed to validate the spec mode")
-		rmnutil.ReportIfNotPresent(v.reconciler.eventRecorder, v.instance, corev1.EventTypeWarning,
-			rmnutil.EventReasonValidationFailed, err.Error())
-
-		msg := "VolumeReplicationGroup mode is invalid"
-		setVRGDataErrorCondition(&v.instance.Status.Conditions, v.instance.Generation, msg)
```

Comment:
Previously we set the condition message to different string without the error. You are mixing refactoring with behavior changes which makes it harder to review or fix later. Please separate refactoring and behavior changes.

Reply:
I combined the condition message and log message because they were so similar.  I used the condition message as it could be considered part of the API, though I think doing so is discouraged.

Reply:
More information is added, but it should help the user understand why the mode is invalid.